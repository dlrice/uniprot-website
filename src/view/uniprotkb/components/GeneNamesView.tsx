import React, { Fragment } from 'react';
import { v1 } from 'uuid';
import { InfoList } from 'franklin-sites';
import { GeneNamesData } from '../../../model/uniprotkb/sections/NamesAndTaxonomyConverter';

export const GeneAlternativeNamesView: React.FC<{
  alternativeNames: { value: string }[];
  firstComma?: boolean;
}> = ({ alternativeNames, firstComma = true }) => (
  <Fragment>
    {alternativeNames.map((altName, i) => (
      <Fragment key={altName.value}>
        {firstComma || (!firstComma && i > 0) ? ', ' : ''}
        {altName.value}
      </Fragment>
    ))}
  </Fragment>
);

const GeneNamesView: React.FC<{
  geneNamesData: GeneNamesData;
  isFlat?: boolean;
}> = ({ geneNamesData, isFlat = false }) => (
  <Fragment>
    {geneNamesData.map(geneNames => {
      const infoData = [
        {
          title: 'Name',
          content: geneNames.geneName && (
            <Fragment>{geneNames.geneName.value}</Fragment>
          ),
        },
      ];
      if (geneNames.synonyms) {
        infoData.push({
          title: 'Synonyms',
          content: (
            <GeneAlternativeNamesView
              alternativeNames={geneNames.synonyms}
              firstComma={false}
            />
          ),
        });
      }
      if (geneNames.orfNames) {
        infoData.push({
          title: 'ORF names',
          content: (
            <GeneAlternativeNamesView
              alternativeNames={geneNames.orfNames}
              firstComma={false}
            />
          ),
        });
      }
      if (geneNames.orderedLocusNames) {
        infoData.push({
          title: 'Ordered locus names',
          content: (
            <GeneAlternativeNamesView
              alternativeNames={geneNames.orderedLocusNames}
              firstComma={false}
            />
          ),
        });
      }
      return (
        <InfoList
          infoData={infoData}
          key={geneNames.geneName ? geneNames.geneName.value : v1()}
          isCompact={isFlat}
          highlightFirstItem={isFlat}
        />
      );
    })}
  </Fragment>
);

export default GeneNamesView;
