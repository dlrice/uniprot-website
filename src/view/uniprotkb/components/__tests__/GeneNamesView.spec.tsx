import React from 'react';
import { render } from 'react-testing-library';
import GeneNamesView from '../GeneNamesView';
import GeneNamesUIDataJson from '../__mocks__/GeneNamesUIData.json';

describe('GeneNames', () => {
  test('should render gene_names', () => {
    const { asFragment } = render(<GeneNamesView {...GeneNamesUIDataJson} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
