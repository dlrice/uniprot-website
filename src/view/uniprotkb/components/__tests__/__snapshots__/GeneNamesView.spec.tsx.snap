// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneNames should render gene_names 1`] = `
<DocumentFragment>
  <div
    class="info-list"
  >
    <div
      class="info-list__item"
    >
      <div
        class="info-list__item__title"
      >
        <h5
          class="bold"
        >
          Name
        </h5>
      </div>
      <div
        class="info-list__item__content"
      >
        some Gene 
        <span
          class="evidence-tag svg-colour-unreviewed"
          data-testid="evidence-tag-trigger"
          role="button"
          tabindex="0"
        >
          <svg
            fill="none"
            height="12"
            viewBox="0 0 10 15"
            width="12"
          >
            <path
              clip-rule="evenodd"
              d="M10 0H0v15h.051L5 10l4.949 5H10V0z"
              fill="#4E5A71"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="evidence-tag__label"
            title=""
          >
            1 automatic annotation
          </span>
        </span>
        <div
          class="evidence-tag-content "
          data-testid="evidence-tag-content"
        >
          <div>
            <h5>
              Automatic assertion according to rules
            </h5>
          </div>
        </div>
      </div>
    </div>
    <div
      class="info-list__item"
    >
      <div
        class="info-list__item__title"
      >
        <h5
          class="bold"
        >
          Synonyms
        </h5>
      </div>
      <div
        class="info-list__item__content"
      >
        some Syn 
        <span
          class="evidence-tag svg-colour-unreviewed"
          data-testid="evidence-tag-trigger"
          role="button"
          tabindex="0"
        >
          <svg
            fill="none"
            height="12"
            viewBox="0 0 10 15"
            width="12"
          >
            <path
              clip-rule="evenodd"
              d="M10 0H0v15h.051L5 10l4.949 5H10V0z"
              fill="#4E5A71"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="evidence-tag__label"
            title=""
          >
            1 automatic annotation
          </span>
        </span>
        <div
          class="evidence-tag-content "
          data-testid="evidence-tag-content"
        >
          <div>
            <h5>
              Automatic assertion according to rules
            </h5>
          </div>
        </div>
      </div>
    </div>
    <div
      class="info-list__item"
    >
      <div
        class="info-list__item__title"
      >
        <h5
          class="bold"
        >
          ORF names
        </h5>
      </div>
      <div
        class="info-list__item__content"
      >
        some orf 
        <span
          class="evidence-tag svg-colour-reviewed"
          data-testid="evidence-tag-trigger"
          role="button"
          tabindex="0"
        >
          <svg
            fill="none"
            height="12"
            viewBox="0 0 10 15"
            width="12"
          >
            <path
              clip-rule="evenodd"
              d="M10 0H0v15h.051L5 10l4.949 5H10V0z"
              fill="#4E5A71"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="evidence-tag__label"
            title=""
          >
            1 publication
          </span>
        </span>
        <div
          class="evidence-tag-content "
          data-testid="evidence-tag-content"
        >
          <div>
            <h5>
              Manual assertion based on experiment
            </h5>
          </div>
        </div>
      </div>
    </div>
    <div
      class="info-list__item"
    >
      <div
        class="info-list__item__title"
      >
        <h5
          class="bold"
        >
          Ordered locus names
        </h5>
      </div>
      <div
        class="info-list__item__content"
      >
        some locus 
        <span
          class="evidence-tag svg-colour-unreviewed"
          data-testid="evidence-tag-trigger"
          role="button"
          tabindex="0"
        >
          <svg
            fill="none"
            height="12"
            viewBox="0 0 10 15"
            width="12"
          >
            <path
              clip-rule="evenodd"
              d="M10 0H0v15h.051L5 10l4.949 5H10V0z"
              fill="#4E5A71"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="evidence-tag__label"
            title=""
          >
            1 automatic annotation
          </span>
        </span>
        <div
          class="evidence-tag-content "
          data-testid="evidence-tag-content"
        >
          <div>
            <h5>
              Automatic assertion according to rules
            </h5>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
